hyco-https/lib/HybridConnectionHttpsServer.js

```diff
@@ -573,52 +576,60 @@ function controlChannelRequest(server, message) {
   // handler to call when the connection sequence completes
   var self = server;
   
-  if ( message.request.method) {
+  if (message.request.method) {
+    // we received a GET request or a small POST http request
+    // the response should be sent over the control channel
     req = new IncomingMessage(message, server.controlChannel);
     if ( message.request.body == true) {
        self.pendingRequest = req;
     } else {
       req.push(null);
     }
-  }
-  
-  // execute the web socket rendezvous with the server
-  try {
-    var client = new WebSocket(address, {
-      headers: headers,
-      perMessageDeflate: false
-    });
-    client.on('open', function() {
-      
-      server.emit('requestchannel', client);
-      if (self.options.clientTracking) {
-        self.clients.push(client);
-        ws.on('close', function() {
-          var index = self.clients.indexOf(client);
-          if (index != -1) {
-            self.clients.splice(index, 1);
-          }
-        });
-      }
 
-      // do we have a request or is this just rendezvous?
-      if ( message.request.method) {
-         var res = new ServerResponse(req);
-         res.requestId = message.request.id;
-         res.assignSocket(client);
-         server.emit('request', req, res);
-      }
-    });
+    var res = new ServerResponse(req);
+    res.requestId = message.request.id;
+    res.assignSocket(server.controlChannel);
+    server.emit('request', req, res);
+  } else {
+    // we received a chunked request or a large (>64kb) request
+    // execute the web socket rendezvous with the server
+    try {
+      var client = new WebSocket(address, {
+        headers: headers,
+        perMessageDeflate: false
+      });
+      client.on('open', function() {
+        
+        server.emit('requestchannel', client);
+        if (self.options.clientTracking) {
+          self.clients.push(client);
+          ws.on('close', function() {
+            var index = self.clients.indexOf(client);
+            if (index != -1) {
+              self.clients.splice(index, 1);
+            }
+          });
+        }
 
-    client.on('error', function(event) {
-      var index = server.clients.indexOf(client);
-      if (index != -1) {
-        server.clients.splice(index, 1);
-      }
-    });
+        // do we have a request or is this just rendezvous?
+        // if ( message.request.method) {
+        //   var res = new ServerResponse(req);
+        //   res.requestId = message.request.id;
+        //   res.assignSocket(client);
+        //   server.emit('request', req, res);
+        // }
```

Comment:
It seems this is all handled above, can this just be deleted?

Reply:
The request get read around line 442 above by the control channel. The same flow will now apply to all the requests, instead of some requiring the rendezvous to be read.

---
In reply to: [315857871](https://github.com/Azure/azure-relay-node/pull/33#discussion_r315857871) [](ancestors = 315857871)

Reply:
Let's talk in person. I don't quite follow. The cloud service gets to decide some requests (chunked or > 64KB) must be read over the rendezvous connection.

---
In reply to: [315951908](https://github.com/Azure/azure-relay-node/pull/33#discussion_r315951908) [](ancestors = 315951908,315857871)